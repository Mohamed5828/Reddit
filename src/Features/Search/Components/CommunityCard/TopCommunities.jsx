import { CommunityContainer } from "./TopCommunities.styled";
import { Link } from "react-router-dom";
import CommunityCardItem from "../CommunityCardItem/CommunityCardItem";

/**
 * Component that contains the Community Card Elements.
 *
 * @Component
 * @param {object} CommunityList -  contains the Community list elements
 * @param {object} CommunitiesSub2 -  contains the Community list elements that user subscribe
 * @returns {React.Component}
 */
const TopCommunities = ({ CommunityList, CommunitiesSub2 }) => {
  console.log(CommunitiesSub2);
  if (CommunitiesSub2 && CommunityList) {
    return (
      <CommunityContainer>
        <p className="title">Communities</p>
        {CommunityList.map((Community) => (
          <CommunityCardItem
            communityIcon={Community.communityIcon}
            key={Community.communityID}
            communityID={Community.communityID}
            communityDescription={Community.communityDescription}
            membersCount={Community.membersCount}
            communityName={Community.communityName}
            isJoined={CommunitiesSub2.find((element) => {
              return element.id === Community.communityID;
            })}
          />
        ))}
        <Link to="/search/communities">
          <p className="more" id="see-more">
            See more communities
          </p>
        </Link>
      </CommunityContainer>
    );
  }
};

export default TopCommunities;
