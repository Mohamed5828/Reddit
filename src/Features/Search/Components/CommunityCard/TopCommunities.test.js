import { shallow } from "enzyme";
import TopCommunities from "./TopCommunities";
describe("Top Communities", () => {
  it("this is a test for Top Communities", () => {
    expect(shallow(<TopCommunities />)).toMatchSnapshot();
  });
});
