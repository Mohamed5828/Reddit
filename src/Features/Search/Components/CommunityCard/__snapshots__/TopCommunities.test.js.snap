// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Top Communities this is a test for Top Communities 1`] = `""`;
